### PR TITLE
chore: deps: update go-libp2p to v0.21

### PR DESCRIPTION
**Pull request by MarcoPolo**
## Proposed Changes

Draft PR to help update lotus to use the latest release of go-libp2p. 

Also adds observability to the resource manager via prometheus so operators can debug blocked resources better. See https://ipfs.marcopolo.io/d/MgmGIjjnk/resource-manager?orgId=1 for a live example.

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green
- [ ] We should merge https://github.com/filecoin-project/go-fil-markets/pull/744 first and we can update the go-fil-markets dep here as well.
- [ ] Merge and release https://github.com/libp2p/go-libp2p-kad-dht/pull/784 (currently referencing this commit directly)

### CODE REVIEW

**Comment by jennijuju** [comment]
@MarcoPolo thank you for opening the draft PR! We noticed that https://github.com/libp2p/go-libp2p/releases/tag/v0.21.0 is now published, im wondering if you are planning to update this branch to the stable release and open PR for review?

**Comment by MarcoPolo** [comment]
> @MarcoPolo thank you for opening the draft PR! We noticed that https://github.com/libp2p/go-libp2p/releases/tag/v0.21.0 is now published, im wondering if you are planning to update this branch to the stable release and open PR for review?
> 
> it'd be great if we can have rcmgr autoscale in lotus & and have lotus nodes exposes opencensus metrics as well

Yup! planning on doing that today :)

I'll probably split this into two PRs. One with just the dep updates and the other that enables the opencensus metrics.

**Comment by MarcoPolo** [comment]
We should merge https://github.com/filecoin-project/go-fil-markets/pull/744 first and we can update the go-fil-markets dep here as well.

**Comment by jennijuju** [comment]
thanks @MarcoPolo 

pinged market maintainer for t he review there, tho we probably should do separate PR for updating the market version!

**Comment by MarcoPolo** [comment]
> thanks @MarcoPolo
> 
> pinged market maintainer for t he review there, tho we probably should do separate PR for updating the market version!

We need to update go-fil-markets before merging this PR since there was a breaking change in go-libp2p-core and the current go-fil-markets wouldn't work with this PR since it updates go-libp2p-core.

You get this error if you try:
```
go: finding module for package github.com/libp2p/go-libp2p-core/mux
github.com/filecoin-project/lotus/node imports
        github.com/filecoin-project/go-fil-markets/retrievalmarket/network imports
        github.com/libp2p/go-libp2p-core/mux: module github.com/libp2p/go-libp2p-core@latest found (v0.19.1), but does not contain package github.com/libp2p/go-libp2p-core/mux
```

File: node/modules/lp2p/rcmgr.go
Comment (MarcoPolo) [inline node/modules/lp2p/rcmgr.go:114]: This registers the resource managers opencensus metrics

**Comment by jennijuju** [comment]
>  there was a breaking change 

ah- ack!

**Comment by jacobheun** [comment]
It looks like go-fil-markets and dht had releases for this change, what else is needed here? We have some overlap with this changeset on Boost.

**Comment by hannahhoward** [comment]
also can we give @MarcoPolo push access on Lotus? I just realized this is from a fork.

**Review by magik6k — APPROVED** [review]
Looks good.

Will merge with tagged markets

**Pull request merged**
